src/profile-logic/profile-data.js

```diff
@@ -1522,3 +1522,111 @@ export function getTreeOrderComparator(
     return compareCallNodes(callNodeA, callNodeB);
   };
 }
+
+/**
+ * This function walks up the from a clicked call node, and tries to find the best
+ * call node to select. This is the root-most call node that is drawn on the same
+ * category path as the clicked call node.
+ */
+export function findBestAncestorCallNode(
+  callNodeInfo: CallNodeInfo,
+  sampleCallNodes: Array<IndexIntoCallNodeTable | null>,
+  sampleCategories: Array<IndexIntoCategoryList | null>,
+  clickedCallNode: IndexIntoCallNodeTable,
+  clickedCategory: IndexIntoCategoryList
+): IndexIntoCallNodeTable {
+  const { callNodeTable } = callNodeInfo;
+  if (callNodeTable.category[clickedCallNode] !== clickedCategory) {
+    return clickedCallNode;
+  }
+
+  // Compute the callNodesOnSameCategoryPath.
+  // Given a call node path with some arbitrary categories, e.g. A, B, C
+  //
+  // A -> A -> B -> B -> C -> C -> C
+  //
+  // This loop will select the leaf-most call nodes that match the leaf call-node's
+  // category. Running the above path through this loop would produce the list:
+  //
+  // "C -> C -> C".
```

Comment:
Nice comment. Maybe add some call node IDs to the comment, so that "C[8] -> C[7] -> C[6]" makes it clear that the list is ordered leaf to root-most?